### PR TITLE
chore: update esbuild dependencies for bundling

### PR DESCRIPTION
**Pull request by WilliamBergamin**
### Summary

This PR update the `esbuild` related dependency and improves support for bundling npm packages.

It also gets ride of the `deps.ts` file. This prevents every hook from importing all the project dependencies when executing.

### Testing

1. Create a new app
2. Use the following `slack.json`
```json
{
  "hooks": {
    "get-hooks": "deno run -q --allow-read --allow-net path/to/deno-slack-hooks/src/mod.ts",
    "get-manifest": "deno run -q --config=deno.jsonc --allow-read --allow-net --allow-env --allow-sys=osRelease path/to/deno-slack-hooks/src/get_manifest.ts",
    "build": "deno run -q --config=deno.jsonc --allow-read --allow-write --allow-net --allow-run --allow-env --allow-sys=osRelease path/to/deno-slack-hooks/src/build.ts",
    "get-trigger": "deno run -q --config=deno.jsonc --allow-read --allow-net --allow-env path/to/deno-slack-hooks/src/get_trigger.ts",
    "check-update": "deno run -q --config=deno.jsonc --allow-read --allow-net path/to/deno-slack-hooks/src/check_update.ts",
    "install-update": "deno run -q --config=deno.jsonc --allow-run --allow-read --allow-write --allow-net path/to/deno-slack-hooks/src/install_update.ts",
    "doctor": "deno run -q --config=deno.jsonc --allow-read --allow-net path/to/deno-slack-hooks/src/doctor.ts"
  }
}

```
4. Import the `import { WebClient } from "npm:@slack/web-api";`
5. The app should bundle properly, but it will fail at runtime since `@slack/web-api` requires `--allow-sys=osRelease` to work.

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/{project_slug}/blob/main/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

**Comment by filmaj** [comment]
The bundling works, but I think this PR now requires exposing env vars to `deno` permissions; if I run e.g. `slak -v manifest` from an app that uses this branch of hooks I get:

```
Requires env access to "DENO_REGISTRY_URL", run again with the --allow-env flag
const JSR_REGISTRY_URL = Deno.env.get("DENO_REGISTRY_URL") ?? "https://jsr.io";
```

The `slack.json` for many of our samples don't expose the `--allow-env` permission. ROSI specifically denies it (and likely we won't be able to adjust that), though ROSI only leverages the `deno-slack-runtime` project and not the `-hooks` project. _Maybe_ we can get away with simply exposing `--allow-env` in samples' `slack.json`? Not sure.

**Comment by WilliamBergamin** [comment]
@filmaj I've updated this PR to get ride of the `deps.ts` file. This allows each hook to only import the dependencies it needs to run, instead of all the dependencies in the project.

This also allows Deno to properly restrict system access like `--allow-env`

**Comment by filmaj** [comment]
Sounds good, tomorrow morning I'll run a few tests on a few different apps - fresh and existing - to see how it runs.

File: src/mod.ts [hunk 1]
```diff
@@ -7,11 +7,11 @@ export const projectScripts = (args: string[]) => {
     "runtime": "deno",
     "hooks": {
       "get-manifest":
-        `deno run -q --config=deno.jsonc --allow-read --allow-net --allow-env https://deno.land/x/${HOOKS_TAG}/get_manifest.ts`,
+        `deno run -q --config=deno.jsonc --allow-read --allow-net --allow-env --allow-sys=onRelease https://deno.land/x/${HOOKS_TAG}/get_manifest.ts`,
```
Comment (WilliamBergamin) [inline src/mod.ts:10]: I think the `--allow-sys` settings here should match the ones from ROSI, but I'm not sure what they are

Comment (filmaj) [reply src/mod.ts:10]: Typo:

```suggestion
        `deno run -q --config=deno.jsonc --allow-read --allow-net --allow-env --allow-sys=osRelease https://deno.land/x/${HOOKS_TAG}/get_manifest.ts`,
```

Comment (filmaj) [reply src/mod.ts:10]: I think we should also add this to the start hook, and we probably need to add it as well to the files within `deno-slack-runtime`. I posted in the ROSI channel about doing so, let's see what shakes out there.

**Review by filmaj — APPROVED** [review]
LGTM! Tested it out with a new app as well as existing app. Note that, assuming ROSI OKs access to this system API, the start hook and deno-slack-runtime also will need updates to allow access to the system `osRelease` API.

**Pull request merged**